core/src/main/scala/kafka/server/ClientQuotaManager.scala

```diff
@@ -139,11 +139,14 @@ class ClientQuotaManager(private val config: ClientQuotaManagerConfig,
    * we need to add a delay of X to W such that O * W / (W + X) = T.
    * Solving for X, we get X = (O - T)/T * W.
    */
-  private def throttleTime(metricValue: Double, config: MetricConfig): Int = {
+  private def throttleTime(clientMetric: KafkaMetric, config: MetricConfig): Int = {
+    // Casting to Rate because we only use Rate in Quota computation
+    val rateMetric = clientMetric.measurable().asInstanceOf[Rate]
```

Comment:
This seems fragile, are we enforcing this somewhere? Either way, I would introduce a private method `measurableAsRate` that would use pattern matching to return `Rate` or throw an exception with the appropriate error message.